File: RhythmTap/AudioTrack.m
//
//  AudioTrack.m
//  RhythmTap
//
//  Created by Brian Yip on 2016-03-06.
//  Copyright © 2016 Brian Yip. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "AudioTrack.h"

@implementation AudioTrack : NSObject 

- (id)init {
    self = [super init];
    self->_file = @"";
    self->_audioFormat = @"";
    return self;
}

- (id)init: (NSString*)file audioFormat:(NSString*)format {
    self = [super init];
    self->_file = file;
    self->_audioFormat = format;
    return self;
}

@end

File: RhythmTap/AudioTrack.swift
//
//  AudioTrack.swift
//  RhythmTap
//
//  Created by Brian Yip on 2016-03-09.
//  Copyright © 2016 Brian Yip. All rights reserved.
//

import Foundation

@objc public class AudioTrack : NSObject {
    
    // MARK: Properties
    var songName: String!
    var audioFormat: String!
    
    override init() {
        self.songName = "";
        self.audioFormat = "";
    }
    
    init(songName: String, audioFormat: String) {
        self.songName = songName
        self.audioFormat = audioFormat
    }
    
    func getFullBundlePath() -> String {
        if let fullpathToFile = NSBundle.mainBundle().pathForResource(songName, ofType:audioFormat) {
            return fullpathToFile
        }
        return "Could not find " + songName + "." + audioFormat + " in main bundle"
    }

}
